### ISSUE TITLE
DEFAULT_TIMEOUT is not passed to Faraday connection as an option

### ISSUE THREAD

**Issue by SeanSmithers** [decision-labs/fcm#84]
When HTTParty was replaced with Faraday (in https://github.com/decision-labs/fcm/pull/55) some HTTParty options were converted to constants (https://github.com/decision-labs/fcm/pull/55/files#diff-0068abed920b95db498a1c507ae5a10f86145599f5dd2431b36ccbf492d7fe7dL6-L10)

`DEFAULT_TIMEOUT` and `FORMAT`  are now unused.

It would be great to pass `DEFAULT_TIMEOUT` as an option to Faraday, example:
```ruby
  def for_uri(uri, extra_headers = {})
    connection = ::Faraday.new(:url => uri) do |faraday|
      faraday.adapter  Faraday.default_adapter
      faraday.options["timeout"] = DEFAULT_TIMEOUT
      faraday.headers["Content-Type"] = "application/json"
      faraday.headers["Authorization"] = "key=#{api_key}"
      extra_headers.each do |key, value|
        faraday.headers[key] = value
      end
    end
    yield connection
  end
```

A bonus would be to allow some way for consumers of `FCM` to be able to pass options like this in when initializing and have them passed to the underlying HTTP library.

**Comment by sabman** [comment]
I am thinking of adding support for `.env` and/or `ENV` variables to overwrite the defaults. Thoughts?

**Comment by SeanSmithers** [comment]
@sabman support for `.env` and/or `ENV` variables could help here, yes.

Would these variables allow a consumer to overwrite just the `timeout` option or would it be possible to overwrite a number of different options on the underlying HTTP library?, e.g. `open_timeout` as well (but perhaps even things like request/response logging)

**Comment by sabman** [comment]
Yes we can probably add other items too. Only thing I'd wanna ensure is avoid accidental name clashes. So maybe namespacing the environment variables with `FCM_VAR_XXX` might be the way to go.

**Comment by jensljungblad** [comment]
@sabman Could you do something similar to the elasticsearch gem?

```ruby
Elasticsearch::Client.new(
  host: ENV.fetch("ELASTICSEARCH_HOST", nil),
  user: ENV.fetch("ELASTICSEARCH_USER", nil),
  password: ENV.fetch("ELASTICSEARCH_PASSWORD", nil),
  request_timeout: 10
) do |faraday|
  faraday.request(:instrumentation, name: "request.elasticsearch")
end
```

Giving access to the faraday object allows you to attach own instrumentation etc.

**Comment by Linuus** [comment]
The issue with `DEFAULT_TIMEOUT` not being used has been addressed in: https://github.com/decision-labs/fcm/pull/96
